src/main/java/de/tum/in/www1/artemis/service/SubmissionPolicyService.java

```diff
@@ -319,6 +319,13 @@ private int getParticipationSubmissionCount(Participation participation) {
         if (submissions != null && !submissions.isEmpty()) {
             submissionCompensation = submissions.iterator().next().getResults().isEmpty() ? 1 : 0;
         }
+        // Note: The way the participation submissions are counted here (filtering out submissions without results), leads to unexpected behavior when the user submits to their
+        // repository in rapid succession.
+        // When the user submits while the result for the previous submission is not yet available, the previous submission will not be counted.
+        // This means that the user is able to submit more often than the allowed number of submissions, if a lock repository policy is configured.
+        // As these submissions have to happen in quick succession, this does not constitute an advantage for the student and the behaviour is acceptable for now.
+        // Nonetheless, this method should be reassessed to determine whether it is also feasible to count the submissions without taking results into account.
+        // TODO: Reassess method to determine whether it is also feasible to count the submissions without taking results into account.
```

Comment:
This is the intended behavior. Students should only loose a submission, when they got an advantage i.e. a result from the build system

Reply:
Removed the TODO